esr/fitting/match.py

```diff
@@ -147,17 +152,70 @@ def f1(x):
 
             except:
                 negloglike_all[i] = np.nan
-
-            k -= np.sum(Nsteps<1)       # For the codelen, we effectively don't have the parameter that had Nsteps<1
                 
+            if np.isfinite(negloglike_all[i]):
+                k -= np.sum(Nsteps<1)
+                kept_mask = Nsteps>=1
+            else:
+                # Let's see if setting any of the parameters to zero is ok
+                try_idx = np.arange(nparams)[Nsteps < 1]
+                for r in reversed(range(1, len(try_idx))):
+                    for idx in itertools.combinations(try_idx, r):
+                        p = np.copy(ptrue)
+                        p[idx] = 0.
+                        if k==1:
+                            eq_numpy = sympy.lambdify([x, a0], eq, modules=["numpy"])
+                            negloglike_all[i] = f1(p)               # Modified here for this variant, but if this doesn't happen it stays the same as the unique eq
+                        else:
+                            all_a = ' '.join([f'a{i}' for i in range(nparams)])
+                            all_a = list(sympy.symbols(all_a, real=True))
+                            eq_numpy = sympy.lambdify([x] + all_a, eq, modules=["numpy"])
+                            negloglike_all[i] = fop(p)
+                        # negloglike_all[i] = f1(p)
+                        if np.isfinite(negloglike_all[i]):
+                            break
+                kept_mask = np.ones(len(p), dtype=bool)
+                if np.isfinite(negloglike_all[i]):
+                    k -= len(idx)
+                    kept_mask[idx] = 0
+                elif not np.isfinite(negloglike_all[i]) and not np.isnan(negloglike_all[i]): # infinite nll
+                    p = ptrue
+                    fish[Nsteps<1] = 12./(p[Nsteps<1]**2) # set uncertainty=parameter in this case
+                    codelen[i] = -k/2.*math.log(3.) + np.sum( 0.5*np.log(fish_special) + np.log(abs(np.array(p))) )
```

Comment:
fish_special does not seem to be defined anywhere, please do this

Reply:
Yes sorry it should be "fish", I forgot to upload that part from my laptop